//  python -m grpc_tools.protoc --python_out=. --grpc_python_out=. -I. func.proto  
syntax = "proto3";

service Coordinator {
    rpc SayHello(HelloRequest) returns (HelloResponse) {}
    rpc HeartBeat(HeartBeatRequest) returns (HeartBeatResponse) {}
}

message HelloRequest {   // worker-> server
    string w_to_s_msg = 1;
}

message HelloResponse {    // server-> worker
    string s_to_w_msg = 1;
    int32 workerID = 2;
}

message HeartBeatRequest {
    int32 workerID = 1;
}

message HeartBeatResponse {
    int32 status = 1;  // 0-无事发生，1-add repo，2-delete repo
}

message AddRepoURLRequest {
}

message AddRepoURLResponse {
    int32 workerID = 1;
    repeated string repos = 2;  // 数组
}

message DeleRepoURLRequest {
}

message DeleRepoURLResponse {
    int32 workerID = 1;
    repeated string repos = 2;
}